onnxruntime/core/mlas/lib/aarch64/ConvSymS8KernelDotLd64.S

```diff
@@ -0,0 +1,653 @@
+        InputChannels length vector (either from the input tensor or a vector of
+        padding values). These are grouped in batches of length KernelSize.
+        These batches are then repeated OutputCount times.
+
+    Filter (x1) - Points to the filter buffer.
+
+    Output (x2) - Points the output buffer.
+
+    KernelSize (x3/x9) - Size of the kernel (most commonly. 3x3=9, 5x5=25).
+
+        If MLAS_CONV_SYM_FLAG_INPUT_DIRECT is set, then kernel size should be 1.
+
+    InputChannels (x4/x7) - Number of input channels.
+
+    OutputChannels (x5) - Number of output channels.
+
+    ChannelCount (x6) - Number of output channels this iteration produces.
+
+    OutputCount (x7) - Number of output elements this iteration produces.
+
+        This implementation requires the count to be no larger than 4.
+
+    PostProcessParams (x8) - Points to the post process parameter block.
+
+    KernelFlags - (w10) Additional flags controlling the operation.
+
+Return Value:
+
+    None.
+
+--*/
+        FUNCTION_ENTRY MlasConvSymS8KernelDotLd64
+
+        stp     d8,d9,[sp,#-.LConvSymFrame_SavedRegisters]!
+        ldr     x8,[sp,#.LConvSymFrame_PostProcessParams]
+        str     d10,[sp,#16]
+        cmp     x7,2                    // OutputCount < 2 ?
+        str     d11,[sp,#24]
+        add     x16,x2,x5               // x16 -> C1
+        str     x19,[sp,#32]
+        lsl     x3,x3,#3                // KernelSize * sizeof(int8_t*)
+        str     x20,[sp,#40]
+        csel    x16,x2,x16,lo           // if OutputCount < 2  x16/C1 -> C0
+        str     x21,[sp,#48]
+        add     x4,x4,3                 // InputChannels align to 4
+        str     x22,[sp,#56]
+        add     x17,x16,x5              // x17 -> C2
+        str     x23,[sp,#64]
+        ldr     x11,[x8,#.LConvSymPostProcessParams_Bias]
+        csel    x17,x16,x17,ls          // if OutputCount <= 2  x17/C2 -> C1
+        bic     x4,x4,3
+        cmp     x7,4                    // OutputCount < 4 ?
+        ldr     w10,[sp,#.LConvSymFrame_KernelFlags]
+        add     x5,x17,x5               // x5 -> C3
+        ldr     x19,[x8,#.LConvSymPostProcessParams_Scale]
+        csel    x5,x17,x5,lo            // if OutputCount < 4  x5/C3 -> C2
+
+        // TODO!! tiptoe around loading biases if we need to support
+        // output channels none divisible by 16
+OutputChannelLoop:
+        ldp     q16,q20,[x11],32        // Init accumulators with biases
+        mov     v17.16b,v16.16b
+        mov     v18.16b,v16.16b
+        ldp     q24,q28,[x11],32
+        mov     v19.16b,v16.16b
+        mov     v21.16b,v20.16b
+        mov     v22.16b,v20.16b
+        mov     v23.16b,v20.16b
+        mov     v25.16b,v24.16b
+        mov     v26.16b,v24.16b
+        mov     v27.16b,v24.16b
+        mov     v29.16b,v28.16b
+        mov     v30.16b,v28.16b
+        mov     v31.16b,v28.16b
+        mov     x9,x3                   // restore KernelSize * sizeof(int8_t*)
+
+KernelSizeLoop:
```

Comment:
>KernelSizeLoop [](http://example.com/codeflow?start=0&length=14)

Why the InputDirect part is removed?

Reply:
it's faster to use qgemm kernel, since we don't even use indirect buffer in this case.  Less branches also good for perf